File: sprinbootProject/project08_oneToonebidirectional/src/main/java/com/sprinboot/project08_oneToonebidirectional/domain/Student.java
package com.sprinboot.project08_oneToonebidirectional.domain;

import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
public class Student {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String email;
    private String password;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id")
    private Address address;
}

File: sprinbootProject/frontend/validationdemo/src/app/student.ts
export class Student {

    private name:string;
    private email:string;
    private password:string;
    private subjects :string[];

    constructor(name:string,email:string,password:string,subjects:string[] = []){
        this.name = name;
        this.email = email;
        this.password = password;
        this.subjects = subjects;
    }
}
